backpack/context.py

```diff
@@ -1,56 +1,104 @@
+"""Context class for BackPACK."""
+from typing import Callable, Iterable, List, Tuple, Type
+
+from torch.nn import Module
+from torch.utils.hooks import RemovableHandle
+
+from backpack.extensions.backprop_extension import BackpropExtension
 from backpack.utils.hooks import no_op
 
 
 class CTX:
-    """
-    Global Class holding the configuration of the backward pass
-    """
+    """Global Class holding the configuration of the backward pass."""
 
-    active_exts = tuple()
-    debug = False
-    extension_hook = no_op
+    active_exts: Tuple[BackpropExtension] = tuple()
+    debug: bool = False
+    extension_hook: Callable[[Module], None] = no_op
+    hook_handles: List[RemovableHandle] = []
 
     @staticmethod
-    def set_active_exts(active_exts):
+    def set_active_exts(active_exts: Iterable[BackpropExtension]) -> None:
+        """Set the active backpack extensions.
+
+        Args:
+            active_exts: the extensions
+        """
         CTX.active_exts = tuple()
         for act_ext in active_exts:
             CTX.active_exts += (act_ext,)
 
     @staticmethod
-    def get_active_exts():
+    def get_active_exts() -> Tuple[BackpropExtension]:
+        """Get the currently active extensions.
+
+        Returns:
+            active extensions
+        """
         return CTX.active_exts
 
     @staticmethod
-    def add_hook_handle(hook_handle):
-        if getattr(CTX, "hook_handles", None) is None:
-            CTX.hook_handles = []
+    def add_hook_handle(hook_handle: RemovableHandle) -> None:
+        """Add the hook handle to internal variable hook_handles.
+
+        Args:
+            hook_handle: the removable handle
+        """
         CTX.hook_handles.append(hook_handle)
 
     @staticmethod
-    def remove_hooks():
+    def remove_hooks() -> None:
+        """Remove all hooks."""
         for handle in CTX.hook_handles:
             handle.remove()
         CTX.hook_handles = []
 
     @staticmethod
-    def is_extension_active(*extension_classes):
+    def is_extension_active(*extension_classes: Type[BackpropExtension]) -> bool:
+        """Returns whether the specified class is currently active.
+
+        Args:
+            *extension_classes: classes to test
+
+        Returns:
+            whether at least one of the specified extensions is active
+        """
         for backpack_ext in CTX.get_active_exts():
             if isinstance(backpack_ext, extension_classes):
                 return True
         return False
```

Comment:
Suggestion: `return any(isinstance(ext, extension_classes) for ext in CTX.get_active_exts())`